connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroDeserializer.scala

```diff
@@ -160,6 +160,14 @@ private[sql] class AvroDeserializer(
         (logicalDataType, catalystType) match {
           case (LongType, LongType) => (updater, ordinal, value) =>
             updater.setLong(ordinal, value.asInstanceOf[Long])
+          case (_, LongType) => avroType.getLogicalType match {
+            case _: TimestampMicros | _: TimestampMillis |
+                 _: LocalTimestampMillis | _: LocalTimestampMicros |
+                 _: TimeMicros | _: TimeMillis => (updater, ordinal, value) =>
```

Comment:
Why do we handle TimeMicros & TimeMillis here?

Reply:
Why don't we handle it here. Logical type TimeMicros & TimeMillis can also convert to long type.

Reply:
Yeah, you are right. It's already handled. Thank you!